### PR TITLE
Weather - Major Cleanup

### PR DESCRIPTION
**Pull request by ulteq**
* Single global on/off switch (that really turns everything off properly)
* Less coupling between code that runs on the client / server
* Simpler update routines
* More cohesive code
* Less module options
* Less network traffic

- [x] Handle fog (syncs perfectly in vanilla)
- [x] Handle rain (syncs perfectly in vanilla)
- [x] Handle waves (syncs perfectly in vanilla)
- [x] Add master switch for the wind simulation

| Rain  | Fog | Waves |
| ------------- | ------------- | ------------- |
| weather_rain | weather_fog | weather_waves |

### CODE REVIEW

File: addons/weather/functions/fnc_updateHumidity.sqf [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /*
  * Author: ACE2 Team
- * Updates GVAR(currentHumidity)
+ * Smoothly updates GVAR(currentHumidity) based time of day and map data
```
Comment (ulteq) [inline addons/weather/functions/fnc_updateHumidity.sqf]: Reminder: Typo

File: addons/weather/functions/fnc_updateTemperature.sqf [hunk 2]
```diff
@@ -1,6 +1,6 @@
 /*
  * Author: ACE2 Team
- * Updates GVAR(currentTemperature) based on the map data
+ * Smoothly updates GVAR(currentTemperature) based time of day and map data
```
Comment (ulteq) [inline addons/weather/functions/fnc_updateTemperature.sqf]: Reminder: Typo

File: addons/weather/functions/fnc_updateServerWeather.sqf [hunk 3]
```diff
@@ -9,12 +9,16 @@
...
  * Example:
- * [] call ace_weather_fnc_updateAceWeather
+ * [] call ace_weather_fnc_updateServerWeather
  *
  * Public: No
  */
 #include "script_component.hpp"
 
+if (!GVAR(enabled)) exitWith {
+    missionNamespace setVariable [QGVAR(currentOvercast), 0, true];
```
Comment (PabstMirror) [inline addons/weather/functions/fnc_updateServerWeather.sqf]: could just do the exitWith in XEH_postServerInit.sqf/ace_settingsInitialized
no point in running a PFEH to set a static value

Comment (ulteq) [reply addons/weather/functions/fnc_updateServerWeather.sqf]: One of the design goals was to be able to switch the weather module on and off during the mission.

Comment (ulteq) [reply addons/weather/functions/fnc_updateServerWeather.sqf]: We could simplify the code in many places, if we drop that objective.

Comment (bux) [reply addons/weather/functions/fnc_updateServerWeather.sqf]: > One of the design goals was to be able to switch the weather module on and off during the mission.

y tho?

Comment (ulteq) [reply addons/weather/functions/fnc_updateServerWeather.sqf]: `GVAR(currentOvercast)` needs to be set to `0` on all clients, when the Weather module is disabled.

Comment (bux) [reply addons/weather/functions/fnc_updateServerWeather.sqf]: no, i mean, why  should anyone want to turn it off mid mission

Comment (ulteq) [reply addons/weather/functions/fnc_updateServerWeather.sqf]: Ok, let's get rid of it.

**Comment by sargken** [comment]
Will this work with the weather module

**Comment by sargken** [comment]
I meant will this work with the Zeus weather module?

**Comment by ulteq** [comment]
> I meant will this work with the Zeus weather module?

I do not know. We have to test it.

File: addons/weather/stringtable.xml [hunk 4]
```diff
@@ -130,10 +118,12 @@
...
         </Key>
         <Key ID="STR_ACE_Weather_windSimulation_DisplayName">
-            <English>Wind Simulation</English>
+            <English>Wind Simulation (map based)</English>
+            <German>Wind Simulation (kartenbasiert)</German>
         </Key>
         <Key ID="STR_ACE_Weather_windSimulation_Description">
-            <English>Enables the wind simulation (map based)</English>
+            <English>Enables the map based wind simulation (overwrites vanilla wind)</English>
+            <German>Aktiviert die kartenbasierte Windsimulation (überschreibt Vanilla Wind)</German>
```
Comment (bux) [inline addons/weather/stringtable.xml]: Vanilla-Wind

**Comment by SilentSpike** [comment]
Something I meant to ask concerning this PR: Are the test results shown using the manual weather options in the mission editor? Ideally we should've checked that it synced whether those are on or off

**Comment by SilentSpike** [comment]
Nevermind, I see Cuel has relevant plots here: https://cuel.github.io 👍

**Pull request merged**